crates/deltalake-core/src/operations/merge/mod.rs

```diff
@@ -1418,9 +1416,7 @@ impl std::future::IntoFuture for MergeBuilder {
             PROTOCOL.can_write_to(&this.snapshot)?;
 
             let state = this.state.unwrap_or_else(|| {
-                //TODO: Datafusion's Hashjoin has some memory issues. Running with all cores results in a OoM. Can be removed when upstream improvemetns are made.
                 let config: SessionConfig = DeltaSessionConfig::default().into();
-                let config = config.with_target_partitions(1);
```

Comment:
can this be removed due to the fix you mentioned? Otherwise @Blajda, would we still require this to be a single partition?

Reply:
Looks like they resolved the HashJoin memory issue. Running the benchmarks on my system with this branch does not result any OoM issues. 

Great to see this finally fixed! Will result a speed for users of this operation.
Before is main
After is this PR

```
+-------------------------------------------------------------------------------------+---------------------+--------------------+------------------------------------------+
| name                                                                                | before_duration_avg | after_duration_avg | before_duration_avg / after_duration_avg |
+-------------------------------------------------------------------------------------+---------------------+--------------------+------------------------------------------+
| delete_only_fileMatchedFraction_0.05_rowMatchedFraction_0.05                        | 2346.0              | 970.0              | 2.4185567010309277                       |
| multiple_insert_only_fileMatchedFraction_0.05_rowNotMatchedFraction_0.05            | 1919.0              | 972.0              | 1.9742798353909465                       |
| multiple_insert_only_fileMatchedFraction_0.05_rowNotMatchedFraction_0.50            | 2002.0              | 1021.0             | 1.960822722820764                        |
| multiple_insert_only_fileMatchedFraction_0.05_rowNotMatchedFraction_1.0             | 1956.0              | 1050.0             | 1.8628571428571428                       |
| upsert_fileMatchedFraction_0.05_rowMatchedFraction_0.01_rowNotMatchedFraction_0.1   | 2163.0              | 1044.0             | 2.07183908045977                         |
| upsert_fileMatchedFraction_0.05_rowMatchedFraction_0.0_rowNotMatchedFraction_0.1    | 1933.0              | 998.0              | 1.93687374749499                         |
| upsert_fileMatchedFraction_0.05_rowMatchedFraction_0.1_rowNotMatchedFraction_0.0    | 2187.0              | 957.0              | 2.2852664576802506                       |
| upsert_fileMatchedFraction_0.05_rowMatchedFraction_0.1_rowNotMatchedFraction_0.01   | 2131.0              | 1024.0             | 2.0810546875                             |
| upsert_fileMatchedFraction_0.05_rowMatchedFraction_0.5_rowNotMatchedFraction_0.001  | 2190.0              | 1063.0             | 2.060206961429915                        |
| upsert_fileMatchedFraction_0.05_rowMatchedFraction_0.99_rowNotMatchedFraction_0.001 | 2221.0              | 999.0              | 2.2232232232232234                       |
| upsert_fileMatchedFraction_0.05_rowMatchedFraction_1.0_rowNotMatchedFraction_0.001  | 2192.0              | 1046.0             | 2.095602294455067                        |
| upsert_fileMatchedFraction_0.5_rowMatchedFraction_0.001_rowNotMatchedFraction_0.001 | 1983.0              | 1025.0             | 1.9346341463414634                       |
| upsert_fileMatchedFraction_1.0_rowMatchedFraction_0.001_rowNotMatchedFraction_0.001 | 2187.0              | 1094.0             | 1.9990859232175502                       |
+-------------------------------------------------------------------------------------+---------------------+--------------------+------------------------------------------+

Reply:
That's quite a speed boost!

Reply:
Wow!